wherehows-backend/app/controllers/Application.java

```diff
@@ -76,4 +81,37 @@ public static Result index() {
   public static Result healthcheck() {
     return ok("GOOD");
   }
+
+  public static Result deploylibsbackend() {
+    String env = System.getenv("WHZ_APP_HOME");
+    String libpath = env + "/lib";
+    String commitfile = env + "/commit";
+    String libraries = "";
+    String commit = "";
+
+    try {
+      BufferedReader br = new BufferedReader(new FileReader(commitfile));
+      while ((commit = br.readLine()) != null){
+
+      }
+    } catch(IOException ioe) {
+      Logger.error("Error while reading commit file. Error message: " +
+              ioe.getMessage());
+    }
+
+    if (env == null)
+    {
+      return ok("wherehows");
+    }
+
+    //get all the files from a directory
+    File directory = new File(libpath);
+    for (File file : directory.listFiles()) {
+      if (file.isFile()) {
+        libraries += file.getName() + "\n";
+      }
+    }
+
+    return ok(commit+"\n"+libraries);
```

Comment:
Add space before and after + . Or just use intellij to reformat your part of code.